src/pose_estimation/pre-processing/keypoint_scaling.py

```diff
@@ -1,3 +1,163 @@
+import math
+from pose_estimation.keypoint_statistics import KeypointStatistics
+from pose_estimation.keypoints import Keypoints
 
-class KeypointScaling():
-    pass
\ No newline at end of file
+class KeypointScaling:
+    @staticmethod
+    def calculate_scaling_factor(ref_landmark, live_landmark):
+        return KeypointScaling.calculate_distance(ref_landmark.left_shoulder, ref_landmark.right_shoulder) /        KeypointScaling.calculate_distance(live_landmark.left_shoulder, live_landmark.right_shoulder)
+
+    @staticmethod
+    def calculate_distance(a, b):
+        return math.sqrt((a.x - b.x) ** 2 + (a.y - b.y) ** 2)
+
+    @staticmethod
+    def scale_keypoint_distance(key_point1, key_point2, scaling_factor):
+        return KeypointScaling.calculate_distance(key_point1, key_point2) * scaling_factor
+
+    @staticmethod
+    def compute_point(x1, y1, x2, y2, theta, k):
+        dx = x1 - x2
+        dy = y1 - y2
+        theta = math.radians(theta)
+        rdx = dx * math.cos(theta) - dy * math.sin(theta)
+        rdy = dx * math.sin(theta) + dy * math.cos(theta)
+        ab = math.sqrt(dx*dx + dy*dy)
+        return (k/ab * rdx + x2, k/ab * rdy + y2)
```

Comment:
also more generally, does rotating here make sense? Do you have a source for this?

Reply:
This is the way Tango did for scaling. But I agree, that rotation can be removed, I will update the code

Reply:
Oh no, my bad. Rotation is essential in this scenario. For instance, if we have scaled the coordinates of the left elbow, and now we intend to scale the left wrist, a direct scaling of the wrist coordinates would be incorrect. This is because the coordinates of the left elbow have changed and we do not know the l_wrist coordinate according to the changed l_elbow coordinate, necessitating the consideration of the angle formed by the left shoulder, left elbow, and left wrist. To ensure the angle remains constant, we must create new scaled coordinates for the left wrist, factoring in the rotation applied to the left shoulder and left elbow segments. The rotation of the left shoulder and left elbow segments is to maintain the desired angle before applying the scaling factor.